src/DataStructures/FixedHashMap.hpp

```diff
@@ -575,7 +578,8 @@ template <size_t MaxSize, class Key, class ValueType, class Hash,
 bool operator==(
     const FixedHashMap<MaxSize, Key, ValueType, Hash, KeyEqual>& a,
     const FixedHashMap<MaxSize, Key, ValueType, Hash, KeyEqual>& b) noexcept {
-  return a.size_ == b.size_ and a.data_ == b.data_;
+  return a.size_ == b.size_ and
+         std::is_permutation(a.data_.begin(), a.data_.end(), b.data_.begin());
```

Comment:
I think the way to optimize this is loop over the keys of `a` and check that the data in `b` is the same.

Reply:
Seems simple enough, pushed a fixup